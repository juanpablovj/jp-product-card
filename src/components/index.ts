import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductImage } from './ProductImage';
import { ProductTitle } from './ProductTitle';
import { ProductButtons } from './ProductButtons';
import { ProductCardHOCProps } from '../interfaces/interfaces';

const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
	Title: ProductTitle,
	Image: ProductImage,
	Buttons: ProductButtons,
});

export { ProductCard, ProductImage, ProductTitle, ProductButtons };
