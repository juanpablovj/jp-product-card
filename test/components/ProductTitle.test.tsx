import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {
  test('should de mostrar el componente correctamente con el titulo personalizado', () => {
    const wrapper = renderer.create(
        <ProductTitle title='El titulo' className='custom-class'/>
    )
    expect(wrapper).toMatchSnapshot();
  });

  test('should de mostrar el componente con el nombre del producto', () => { 
    const wrapper = renderer.create(
        <ProductCard product={product1} className='custom-class'>
            {() => <ProductTitle />}
        </ProductCard>
    )
    expect(wrapper).toMatchSnapshot();
   })
});
