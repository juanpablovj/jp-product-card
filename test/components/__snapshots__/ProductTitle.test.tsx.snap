// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should de mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard custom-class"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Sin imagen
  </span>
</div>
`;

exports[`ProductTitle should de mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription custom-class"
>
  El titulo
</span>
`;
